src/xcfun.cpp

```diff
@@ -634,6 +635,106 @@ void xc_eval(xc_functional_obj *f, const double *input, double *output)
     }
 }
 
+int xc_user_eval_setup(xc_functional fun,
+                       const int order,
+                       const unsigned int func_type, // LDA (0), GGA (1), metaGGA (2), taylor (3)
+                       const unsigned int dens_type,  // A (0), N (1), A_B (2), N_S (3)
+                       const unsigned int mode_type,  // same as the enum list
+                       const unsigned int laplacian,  // 0/1 laplacian no/yes 
+                       const unsigned int kinetic,    // 0/1 kinetic energy no/yes
+                       const unsigned int current,    // 0/1 current density no/yes
+                       const unsigned int explicit_derivatives) {   // 0/1 gamma vs explicit partial derivatives 
+
+    if (func_type > 3 || dens_type > 3 || mode_type > 3 || laplacian > 1 || kinetic > 1 || current > 1 || explicit_derivatives > 1) {
+        xcint_die("xc_user_eval_setup: invalid input",-1);
+    }
+    
+
+    enum xc_vars vars = XC_VARS_UNSET; 
+    enum xc_mode mode = XC_MODE_UNSET;
+    
+    switch (mode_type){
+    case(1): mode = XC_PARTIAL_DERIVATIVES; break;
+    case(2): mode = XC_POTENTIAL;           break;
+    case(3): mode = XC_CONTRACTED;          break;
+    default:
+        xcint_die("xc_user_eval_setup: Invalid mode", mode_type);
+    }
+
+
+    /* Bit encoding of information 
+       7   6   5   4   3   2   1   0
+    ------------------------------------------------------------------
+       0   0                             LDA
+       0   1                             GGA
+       1   0                             metaGGA
+       1   1                             Taylor
+    ------------------------------------------------------------------
+               0   0                     alpha density
+               0   1                     n density
+               1   0                     anpha and beta densities
+               1   1                     n and s density
+    ------------------------------------------------------------------
+                       0                 no laplacian 
+                       1                 laplacian required
+    ------------------------------------------------------------------
+                           0             no kinetic energy
+                           1             kinetic energy required
+    ------------------------------------------------------------------
+                               0         no current density required
+                               1         current density required
+    ------------------------------------------------------------------
+                                   0     gamma-type partial derivatives
+                                   1     explicit partial derivatives
+     */
```

Comment:
And a `// clang-format on` at the end of the documentation block. This documentation block should go into the docs, but that's not a pressing concern right now.